projects/igniteui-angular/src/lib/grids/api.service.ts

```diff
@@ -150,18 +150,14 @@ export class GridBaseAPIService <T extends IgxGridBaseDirective & GridType> {
             return args;
```

Comment:
Additional improvement could be if we change the `get_row_index_in_data` method. Now it is calling `get_all_data` in its body. But we already has the `data`. Please add second, optional parameter to `get_row_index_in_data` - named `data`. In the method body if `data` is provided use it, otherwise go and get it.

Reply:
done